src/jit/morph.cpp

```diff
@@ -9190,7 +9190,7 @@ GenTree* Compiler::fgMorphRecognizeBoxNullable(GenTree* compare)
         return compare;
     }
 
-    if (opCns->gtIntConCommon.IconValue() != 0)
+    if (!opCns->IsZero())
```

Comment:
I disagree that this is a bad thing - because the conservatism is minor, while adding many flavors of Zero just pollutes the API.